spec/lang/representation.md

```diff
@@ -267,13 +266,14 @@ impl Type {
     fn encode<M: Memory>(Type::Enum { variants, .. }: Self, val: Value<M>) -> List<AbstractByte<M::Provenance>> {
         let Value::Variant { idx, data } = val else { panic!() };
 
-        // idx is to be guaranteed in bounds by the well-formed check in the type.
+        // `idx` is guaranteed to be in bounds by the well-formed check in the type.
         let Variant { ty: variant, tagger } = variants[idx];
         let mut bytes = variant.encode(data.extract());
 
-        // write tag afterwards into the encoded bytes. This is fine as we don't allow
-        // encoded data and the tag to overlap at the moment.
+        // Write tag into the bytes around the data.
+        // This is fine as we don't allow encoded data and the tag to overlap.
         for (offset, value) in tagger.iter() {
+            assert_eq!(Some(AbstractByte::Uninit), bytes.get(offset.bytes()));
```

Comment:
For some reason it doesn't compile when I use `bytes[offset.bytes()]` even though I am indexing in a `List` using an `Int`

Reply:
Otherwise this PR is ready for review again.

Reply:
> doesn't compile

What's the error message? Don't just say "it doesn't work", always explain what happens instead. :)

Reply:
"cannot index into a value of type `libspecr::List<mem::AbstractByte<<M as mem::Memory>::Provenance>>`".
Reading further into this, apparently specr doesn't transpile correctly in the case of `assert_eq!(AbstractByte::Uninit, bytes[offset.bytes()]);`. I noted this for myself to maybe fix it in the future, should I make an issue in the minirust-tooling repo?

Reply:
Ah, is the macro getting in the way? That's possible. 

Sure, please file an issue! This will likely be non-trivial to fix though, and it's not very important.

Reply:
Changed everything else as requested.